packages/auth/src/core/auth/auth_impl.ts

```diff
@@ -221,9 +220,42 @@ export class AuthImpl implements AuthInternal, _FirebaseService {
     await this._updateCurrentUser(user, /* skipBeforeStateCallbacks */ true);
   }
 
+  private async loadUserFromIdToken(idToken: string): Promise<void> {
+    try {
+      const response = await getAccountInfo(this, { idToken });
+      const user = await UserImpl._fromGetAccountInfoResponse(
+        this,
+        response,
+        idToken
+      );
+      await this.directlySetCurrentUser(user);
```

Comment:
Why use `directlySetCurrentUser` instead of `updateCurrentUser`? `directlySetCurrentUser` is rarely used.

Reply:
I noted that `directlySetCurrentUser` was what was being used elsewhere in `initializeCurrentUser` where I am calling this from. FWIW we have chosen to disable `updateCurrentUser` in FirebaseServerApp. I could use `_updateCurrentUser` as @DellaBitta did before but I was concerned that the side effects of notifying listeners and running middleware were undesirable during auth initialization and may lead to more race conditions in developers code and ill effects.

Reply:
Perhaps I should rename this method, as it is more appropriately `initializeCurrentUserFromIdToken`

Reply:
seconded the rename.

Reply:
Renamed.

Reply:
Yeah, the concern of side effects of `_updateCurrentUser` makes sense to me. Thanks, let's keep `directlySetCurrentUser` then.